### DISCUSSION TITLE
Unable to see agent on meshcentral server

### DISCUSSION THREAD

**Discussion by pvssrikanth** [Ylianst/MeshCentral#6007] [Issues]
Hi Team,
I have installed meshcentral server using docker image and server is running fine on my.domain.com 

I have downloaded agent and installed on windows machine 
![image](https://github.com/Ylianst/MeshCentral/assets/17564499/d4b84d8d-2ba5-4f28-90d9-bb983b90b771)

after running agent , it shows its running fine in vm-group-a

But on server  in this group it is NOT showing 

kindly guide how i can fix this issue

I have added hosts file in windows and route ping all is working fine w.r.t connectivity.

Please reply what else i have to do in configuration 

thanks
srikanth

**Comment by mtran-88** [comment]
Hi, how did you install the agent?  Did you go through the group, and select the 'add agent'?
![image](https://github.com/Ylianst/MeshCentral/assets/166559124/31c1305a-2d48-49ec-81c7-3ee0248eca2d)

After downloading the agent, I assume you installed it on the client.   Let me know and I can try to assist.  Cheers.

**Comment by si458** [comment]
@pvssrikanth without seeing ur config.json, it's really hard to tell.
like for example, the is a bug if u use sessionrecording, that after so long, devices don't get added until u restart ur meshcentral server!?

**Comment by pvssrikanth** [comment]
Hi Team
It got resolved after editing msh file with proper server URL it is showing
wss local
After editing server URL it got detected

Can agent running outside connect to meshcental running in LAN

Or do I have to make it public accessible with public ip ?


Thanks for your support

**Comment by si458** [comment]
If the url is showing as `local` then ur meshcentral is running in LAN mode, and will only work when the remote computers are connected to the same network as your meshcentral.

To get it to work in LAN+WAN mode, you need to set `cert` in your config.json.

But again without seeing ur current config.json we can't really know how you are running it

**Comment by pvssrikanth** [comment]
Hi Ylianst/MeshCentral ,

Please find attached docker json file

Please suggest what I have to do changes so that meshcentral running in Lan
, I should be able to install clients in WAN and mesh client should be able
to connect to meshcentral server running in LAN.

Thanking you,
srikanth

**Reply by mtran-88** [reply]
In order for you to access the devices on meshcentral located in the LAN, you would need to have the server be in Hybrid Mode which means it can be on LAN and WAN.  The server has to have a Fixed DNS name. i.e. meshcentral.mydomain.com. This can be changed in the config.json file.  You would need to create an A record for this with a public IP address on your domain.com, and then NAT that to the server on your router.  Hope that helps.   In addition, once you set the cert to a fixed DNS name, the WANonly and LANonly will be set to 'True.

See sample below:

    "cert": "meshcentral.yourdomain.com",
    "_WANonly": true,
    "_LANonly": true,

**Comment by pvssrikanth** [comment]
Awesome, thanks for the info!

I will do these configuration changes and update

Thanks for your support

**Comment by pvssrikanth** [comment]
Hi Ylianst/MeshCentral

I have one requirement in meshcental

After connecting agent
I want to see all applications software also like Adobe  msoffice  ssms
etc.

As of now meshcental dashboard is showing only operating system antivirus
and few

It will be great if all softwares installed are displayed which acts like
asset inventory

Please suggest and advise

Thanks
Srikanth

**Comment by pvssrikanth** [comment]
Hi Ylianst/MeshCentral *Sir*

I kept public ip dns mapping and tried to generate ssl certificate using
letsencrypt in docker-compose file but getting errors

meshcentral    | MeshCentral HTTP redirection server running on port 80.
meshcentral    | Error: abc.com does not match name in TLS certificate:
my.domain.com, localhost
meshcentral    | Generating certificates, may take a few minutes...
meshcentral    | Generating HTTPS certificate...
meshcentral    | Generating Intel AMT MPS certificate...
meshcentral    | MeshCentral v1.1.22, LAN mode, Production mode.
meshcentral    | Code signed MeshCmd64.exe.
meshcentral    | Code signed MeshCmdARM64.exe.
meshcentral    | Code signed MeshCmd.exe.
meshcentral    | Code signed MeshService64.exe.
meshcentral    | Code signed MeshServiceARM64.exe.
meshcentral    | Code signed MeshService.exe.
meshcentral    | Failed to load web certificate at: "https://abc.com",
host: "abc.com"
meshcentral    | MeshCentral HTTPS server running on port 443, alias port
443.

Kindly advice how to make this up using ssl encrption https://abc.com

thanks
srikanth

**Reply by si458** [reply]
you need to share your config.json for us to help you

**Comment by pvssrikanth** [comment]
Hi Ylianst/MeshCentral ,
Thanks for your prompt reply
Delay from my side in sharing the config.json.

Please find attached config.json.

I want guidance on following please
1. How to enable ssl letsencrypt ssl certificate for domain
https://itops.mediamint.com, DNS configuration is already made for domain
itops.mediamint.com
2. From outside when i am accessing itops.mediamint.com its throwing error
reconnect,,, But same is working in LAN environment on domain my.domain.com

Thanks for your excellent support on this excellent meshcentral tool (
Awesome tool )

How many systems in Dashboard we can connect and see any inputs on limit or
only we have to keep BIG TV Screen to see many

Looking forward for your reply.
Srikanth

**Comment by pvssrikanth** [comment]
Sorry attached the config.json

I am running meshcentral on docker if this also has to be changed
attached docker compose file as well.

Thanks

**Comment by pvssrikanth** [comment]
Hi Sir,
I am getting following error while logging in from outside WAN.



Do we have to create NEW Group meshagent for public access and install [image:
image.png]

**Comment by mtran-88** [comment]
Hi, where is the config.json information?  Just as a note, do NOT set the 'production' to 'True' until you test your domain with Let's encrypt and everything passes.  You will get locked out if too many tries.  Not sure what to do if that happens except contact Let's encrypt support. 
 This is what I got when I've tried your domain.  You would need to fix that before setting the server to production.  Hope that helps.  

https://letsdebug.net/itops.mediamint.com/1971933
![image](https://github.com/Ylianst/MeshCentral/assets/166559124/f5e474b2-8e91-4441-ae91-b5a98fc0e844)

**Comment by pvssrikanth** [comment]
Hi Sir,

Thanks for your prompt response.

Attached config.json and docker-compose.yaml

Please guide how I should fix this config.json configuration changes

Thanks

**Reply by si458** [reply]
github isnt getting the attachments, please upload them using the web ui instead

**Comment by pvssrikanth** [comment]
Hi Simon,

Kindly help on this last issue.

We are able to remote access MAC Laptop.

However when we want to remote control ( open application for remote
management). It is NOT allowing.

Kindly help on this last issue for fix so that MESH central is a great TOOL
and we will spread this sweet tool to all industry.

Thanks & Regards

**Reply by si458** [reply]
https://ylianst.github.io/MeshCentral/meshcentral/faq/#mac-clients
